pkg/credentials/config.go

```diff
@@ -113,18 +113,24 @@ func (cp *CredentialsProvider) getExecConfigFromConfig(providerName string) *cli
 	return nil
 }
 
-func (cp *CredentialsProvider) getProviderFromClusterProfile(cluster *v1alpha1.ClusterProfile) *v1alpha1.CredentialProvider {
-	cpProviderTypes := map[string]*v1alpha1.CredentialProvider{}
+func (cp *CredentialsProvider) getClusterAccessFromClusterProfile(cluster *v1alpha1.ClusterProfile) *v1alpha1.AccessProvider {
+	accessProviderTypes := map[string]*v1alpha1.AccessProvider{}
 
-	for _, provider := range cluster.Status.CredentialProviders {
-		newProvider := provider.DeepCopy()
-		cpProviderTypes[provider.Name] = newProvider
+	// to keep backward compatibility, we first check CredentialProviders
+	for _, accessProvider := range cluster.Status.CredentialProviders {
+		newProvider := accessProvider.DeepCopy()
+		accessProviderTypes[accessProvider.Name] = newProvider
 	}
 
-	// we return the first provider that the CP supports.
+	for _, accessProvider := range cluster.Status.AccessProviders {
+		newProvider := accessProvider.DeepCopy()
+		accessProviderTypes[accessProvider.Name] = newProvider
+	}
+
+	// we return the first access provider that the CP supports.
 	for _, providerType := range cp.Providers {
-		if provider, found := cpProviderTypes[providerType.Name]; found {
-			return provider
+		if accesser, found := accessProviderTypes[providerType.Name]; found {
```

Comment:
should accesser should be accessor everywhere?
```suggestion
		if accessor, found := accessProviderTypes[providerType.Name]; found {
```